fppanalysis/deconvolution_methods.py

```diff
@@ -42,330 +44,222 @@ def RL_gauss_deconvolve(sig, kern, iterlist, init=None, cutoff=1e-10, sf=1):
-    # If signconv is negative then b = cutoff - sf*np.amin(sigconv). Cutoff avoids division by 0.
-    # If sigconv is positive, let b = cutoff to ensure non-zero division.
-    if np.amin(sigconv) < 0:
-        b_min = np.amin(sigconv)
-        b = cutoff - sf * b_min
+    error = np.zeros(iteration_list[-1] + 1)
+    error[0] = np.sum(
+        (signal_temporary - fftconvolve(current_result, kern_temporary, "same")) ** 2
+    )
+    result = np.zeros([signal.size, len(iteration_list)])
+
+    inverse_kern = kern_temporary[::-1]
+    signal_convolution_inverse_kern = fftconvolve(
+        signal_temporary, inverse_kern, "same"
+    )
+    kern_convolution_inverse_kern = fftconvolve(kern_temporary, inverse_kern, "same")
+
+    # To ensure we have non-negative iterations we apply a condition which is dependent on signal_convolution_inverse_kern
+    # If signal_convolution_inverse_kern is negative then b = cutoff - scale_factor*np.amin(signal_convolution_inverse_kern).
+    # Cutoff avoids division by 0.
+    # If signal_convolution_inverse_kern is positive, let b = cutoff to ensure non-zero division.
+    if np.amin(signal_convolution_inverse_kern) < 0:
+        b_min = np.amin(signal_convolution_inverse_kern)
+        b = cutoff - scale_factor * b_min
     else:
         b = cutoff
 
-    index_array = np.arange(sigtmp.size)
     count = 0
 
-    for i in tqdm(range(1, iterlist[-1] + 1), position=0, leave=True):
+    for i in tqdm(range(1, iteration_list[-1] + 1), position=0, leave=True):
         # If an element in the previous iteration is very close to zero,
         # the same element in the next iteration should be as well.
         # This is handeled numerically by setting all elements <= cutoff to 0
         # and only performing the interation on those elements > cutoff.
 
-        tmp = fftconvolve(update0, kernconv, "same")
+        updated_convolution = fftconvolve(
+            current_result, kern_convolution_inverse_kern, "same"
+        )
+
+        updated_result = (current_result * (signal_convolution_inverse_kern + b)) / (
+            updated_convolution + b
+        )
 
-        update1 = (update0 * (sigconv + b)) / (tmp + b)
+        error[i] = np.sum(
+            (signal_temporary - fftconvolve(updated_result, kern_temporary, "same"))
+            ** 2
+        )
 
-        err[i] = np.sum((sigtmp - fftconvolve(update1, kerntmp, "same")) ** 2)
-        update0[:] = update1[:]
+        current_result[:] = updated_result[:]
 
-        if i == iterlist[count]:
-            print("i = {}".format(iterlist[count]), flush=True)
-            res[:, count] = update1[:]
+        if i == iteration_list[count]:
+            print("i = {}".format(iteration_list[count]), flush=True)
+            result[:, count] = updated_result[:]
             count += 1
 
-    return res, err
+    return result, error
 
 
-def find_amp_ta_3nn(D, tb=None, **kwargs):
+def gpu_RL_gauss_deconvolve(
```

Comment:
To me it seems like to a good idea to combine this function with `RL_gauss_deconvolve()` since most of the code of the gpu function is just copied from there.